### ISSUE TITLE
OAuthLoginButtons not synced with Meteor.settings.OAuth

### ISSUE THREAD

**Issue by ayhid** [cleverbeagle/pup#302]
I do not know if it should be working this way, but on a new project I just needed email/password login so I disabled 

Oauth settings like so

```
{
...
  "private": {
   ...
    "OAuth": {
      "facebook": {
        "enabled": false,
        "appId": "",
        "secret": "",
        "loginStyle": "popup"
      },
      "google": {
        "enabled": false,
        "clientId": "",
        "secret": "",
        "loginStyle": "popup"
      },
      "github": {
        "enabled": false,
        "clientId": "",
        "secret": "",
        "loginStyle": "popup"
      }
    },
   ...
  }
}
```
I assumed that it should be enough to disable login button but it seems that is not managed this way. (see attached screenshot
![pup-login](https://user-images.githubusercontent.com/235381/59697981-1100ff00-91ef-11e9-9ee6-98e23ff3c86a.png)
)

Is there something I missed to do here ?

[Update]
@cleverbeagle after looking at the `OAuthButtons` component code I think that some bits could be refactored.

for example: `api/OAuth/queries.js` instead of: 
```js
import { ServiceConfiguration } from 'meteor/service-configuration';

export default {
  oAuthServices: (parent, args) => {
    const verifiedServices = [];
    args.services.forEach((service) => {
      const serviceConfig = ServiceConfiguration.configurations.findOne({ service });
      if (serviceConfig && serviceConfig.enabled) {
        verifiedServices.push(service);
      }
    });
    console.log('serviceConfig.enabled', verifiedServices);
    return verifiedServices.sort();
  },
};
```
could be:

```js
import { ServiceConfiguration } from 'meteor/service-configuration';

export default {
  oAuthServices: () =>
    ServiceConfiguration.configurations
      .find({ enabled: true }, { sort: { service: 1 } })
      .map((document) => document.service),
};
```
this said, I don't see where this query is used on the component. It's added [here](https://github.com/cleverbeagle/pup/blob/0199e0b767649f8b31152c4bff100e3248117b51/ui/components/OAuthLoginButtons/index.js#L27)  but never used on the component itself

[Update]

I summed up the refactor to enable syncing oAuthButtons with the settings in this PR #305 hope it suits you.

I am using this modification in a project where I did not need other logins than email, and I did not want to do a dirty hack to disable other login services.

**Comment by ayhid** [comment]
@cleverbeagle after looking at the `OAuthButtons` component code I think that some bits could be refactored.

for example: `api/OAuth/queries.js` instead of: 
```js
import { ServiceConfiguration } from 'meteor/service-configuration';

export default {
  oAuthServices: (parent, args) => {
    const verifiedServices = [];
    args.services.forEach((service) => {
      const serviceConfig = ServiceConfiguration.configurations.findOne({ service });
      if (serviceConfig && serviceConfig.enabled) {
        verifiedServices.push(service);
      }
    });
    console.log('serviceConfig.enabled', verifiedServices);
    return verifiedServices.sort();
  },
};
```
could be:

```js
import { ServiceConfiguration } from 'meteor/service-configuration';

export default {
  oAuthServices: (parent, args) => {
     return ServiceConfiguration.configurations.find({ enabled: true }, {sort: {service:1}}).fetch();
  },
};
```

what do you think?

**Comment by cleverbeagle** [comment]
Nice refactor, dig it. The lack of use is likely an omission on my part and needs to be added.

**Comment by thebarty** [comment]
just stumbled upon this myself. Please merge this in for future users

**Comment by cleverbeagle** [comment]
Resolved in 2.1.0. Closing er' up.
